src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Tree/Visitors/ExprVisitorBase.cs

```diff
@@ -582,131 +466,64 @@ private void VisitChildren(Expr pExpr)
-                    (pExpr as ExprArrayInit).OptionalArgumentDimensions = exprRet as Expr;
-                    goto VISIT_BASE_EXPRARRINIT;
-
-                case ExpressionKind.EK_NOOP:
-                    goto VISIT_EXPRNOOP;
-                VISIT_BASE_EXPRNOOP:
-                    goto VISIT_EXPRSTMT;
-                VISIT_EXPRNOOP:
-                    goto VISIT_BASE_EXPRNOOP;
+                    (pExpr as ExprArrayInit).OptionalArgumentDimensions = exprRet;
+                    break;
 
                 case ExpressionKind.EK_BOUNDLAMBDA:
-                    goto VISIT_EXPRBOUNDLAMBDA;
-                VISIT_BASE_EXPRBOUNDLAMBDA:
-                    goto VISIT_EXPR;
-                VISIT_EXPRBOUNDLAMBDA:
                     exprRet = Visit((pExpr as ExprBoundLambda).OptionalBody);
                     (pExpr as ExprBoundLambda).OptionalBody = exprRet as ExprBlock;
-                    goto VISIT_BASE_EXPRBOUNDLAMBDA;
+                    break;
 
+                case ExpressionKind.EK_LOCAL:
+                case ExpressionKind.EK_THISPOINTER:
+                case ExpressionKind.EK_TYPEORNAMESPACE:
+                case ExpressionKind.EK_CLASS:
+                case ExpressionKind.EK_FUNCPTR:
+                case ExpressionKind.EK_MULTIGET:
+                case ExpressionKind.EK_WRAP:
+                case ExpressionKind.EK_NOOP:
                 case ExpressionKind.EK_UNBOUNDLAMBDA:
-                    goto VISIT_EXPRUNBOUNDLAMBDA;
-                VISIT_BASE_EXPRUNBOUNDLAMBDA:
-                    goto VISIT_EXPR;
-                VISIT_EXPRUNBOUNDLAMBDA:
-                    goto VISIT_BASE_EXPRUNBOUNDLAMBDA;
-
                 case ExpressionKind.EK_HOISTEDLOCALEXPR:
-                    goto VISIT_EXPRHOISTEDLOCALEXPR;
-                VISIT_BASE_EXPRHOISTEDLOCALEXPR:
-                    goto VISIT_EXPR;
-                VISIT_EXPRHOISTEDLOCALEXPR:
-                    goto VISIT_BASE_EXPRHOISTEDLOCALEXPR;
-
                 case ExpressionKind.EK_FIELDINFO:
-                    goto VISIT_EXPRFIELDINFO;
-                VISIT_BASE_EXPRFIELDINFO:
-                    goto VISIT_EXPR;
-                VISIT_EXPRFIELDINFO:
-                    goto VISIT_BASE_EXPRFIELDINFO;
-
                 case ExpressionKind.EK_METHODINFO:
-                    goto VISIT_EXPRMETHODINFO;
-                VISIT_BASE_EXPRMETHODINFO:
-                    goto VISIT_EXPR;
-                VISIT_EXPRMETHODINFO:
-                    goto VISIT_BASE_EXPRMETHODINFO;
+                    break;
+
+                default:
+                    pExpr.AssertIsBin();
+                    exprRet = Visit((pExpr as ExprBinOp).OptionalLeftChild);
+                    (pExpr as ExprBinOp).OptionalLeftChild = exprRet;
+                    exprRet = Visit((pExpr as ExprBinOp).OptionalRightChild);
+                    (pExpr as ExprBinOp).OptionalRightChild = exprRet;
```

Comment:
Should probably consolidate e.g. the multiple casts to `ExprBinOp` into one local, but keeping this PR as just simplifying the paths taken without any other changes.